proto/bpfd.proto

```diff
@@ -7,27 +7,71 @@ service Bpfd {
     rpc Unload (UnloadRequest) returns (UnloadResponse);
     rpc List (ListRequest) returns (ListResponse);
     rpc PullBytecode (PullBytecodeRequest) returns (PullBytecodeResponse);
+    rpc Get (GetRequest) returns ( GetResponse );
 }
 
+/* BytecodeImage represents an eBPF program that is packaged and contained within
+ * an OCI container image.
+ */
+
 message BytecodeImage {
     string url = 1;
     int32 image_pull_policy = 2;
-    string username = 3;
-    string password = 4;
+    optional string username = 3;
+    optional string password = 4;
 }
 
-message LoadRequestCommon {
+/* BytecodeLocation is either:
+ * - Parameters to pull an eBPF program stored in an OCI container image.
+ * - Local file path for an image.
+ */
+message BytecodeLocation { 
     oneof location {
-        BytecodeImage image = 1;
-        string file = 2;
-    } ;
-    string name = 3;
-    uint32 program_type = 4;
-    map<string,string> metadata = 5;
-    map<string, bytes> global_data = 6;
-    optional uint32 map_owner_id = 7;
+        BytecodeImage image = 2;
+        string file = 3;
+    }
+}
+
+/* KernelProgramInfo represents the state for a single ITS_BPF_NOT_EBPF program that is
+ * maintained internally by the Linux kernel within its bpf_prog_info
+ * structure.
+ */
+
+message KernelProgramInfo { 
+    uint32 id = 1;
+    /* Same a Name entered by user, but truncated to 16 byte by kernel */
```

Comment:
OK, tweaked some.
```
    /* Name as seen by the kernel. May be truncated version of user enter */
    /* name since limited to 16 byte by kernel */
```